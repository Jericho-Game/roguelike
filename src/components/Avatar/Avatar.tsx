import classnames from 'classnames';

type OwnProps = Pick<User, 'first_name' | 'second_name' | 'avatar'> & {
  className?: string,
};

export default function Avatar({
  className = '',
  ...user
}: OwnProps) {
  const initials = user.first_name[0] + user.second_name[1];
  return (
    <div
      className={classnames(
        className,
        'flex items-center justify-center overflow-hidden bg-gray-100 rounded-full relative w-1/2 aspect-square',
      )}
    >
      {user.avatar
        ? <img src={user.avatar} alt="avatar" className="m-0" />
        : (<span className="uppercase">{initials}</span>)}
    </div>
  );
}
