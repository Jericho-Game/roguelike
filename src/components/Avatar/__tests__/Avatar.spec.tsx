import { render } from '@testing-library/react';
import Avatar from '../Avatar';

const defaultProps = {
  firstName: 'John',
  secondName: 'Doe',
};

const create = (props = {}) => <Avatar {...defaultProps} {...props} />;

describe('Avatar Test', () => {
  it('matches snapshot', () => {
    const { baseElement } = render(create());

    expect(baseElement).toMatchSnapshot();
  });

  it('with src', () => {
    const { baseElement } = render(create({
      src: './qwe.jpg',
    }));

    expect(baseElement).toMatchSnapshot();
  });

  it('with className', () => {
    const { baseElement } = render(create({
      className: 'self-center text-7xl',
    }));

    expect(baseElement).toMatchSnapshot();
  });
});
