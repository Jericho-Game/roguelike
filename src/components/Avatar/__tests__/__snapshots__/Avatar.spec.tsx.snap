// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Test matches snapshot 1`] = `
<body>
  <div>
    <div
      class="flex items-center justify-center overflow-hidden bg-gray-100 rounded-full relative w-1/2 aspect-square"
    >
      <span
        class="uppercase"
      >
        JD
      </span>
    </div>
  </div>
</body>
`;

exports[`Avatar Test with className 1`] = `
<body>
  <div>
    <div
      class="self-center text-7xl flex items-center justify-center overflow-hidden bg-gray-100 rounded-full relative w-1/2 aspect-square"
    >
      <span
        class="uppercase"
      >
        JD
      </span>
    </div>
  </div>
</body>
`;

exports[`Avatar Test with src 1`] = `
<body>
  <div>
    <div
      class="flex items-center justify-center overflow-hidden bg-gray-100 rounded-full relative w-1/2 aspect-square"
    >
      <img
        alt="John Doe"
        class="m-0"
        src="./qwe.jpg"
      />
    </div>
  </div>
</body>
`;
